sopel/test_tools.py

```diff
@@ -1,12 +1,18 @@
 # coding=utf-8
-"""This module has classes and functions that can help in writing tests.
+"""This module provided tools that helped to write tests.
 
-.. note::
+.. deprecated:: 7.1
+
+    This module is replaced by the
+    :mod:`pytest plugin <sopel.tests.pytest_plugin>` for Sopel; it will be
+    **removed in Sopel 8**.
+
+    It contains mock classes for bot, bot wrapper, and config.
 
    This module formerly contained mock classes for bot, bot wrapper, and config
    objects. Those are deprecated, and will be removed in Sopel 8.0. New code
-   should use the new :mod:`.mocks`, :mod:`.factories`, and
-   :mod:`.pytest_plugin` added in Sopel 7.0.
+   should use the pytest plugin directly; or should take advantage of the
+   :mod:`.mocks` and :mod:`.factories` modules, both added in Sopel 7.0.
```

Comment:
Is this as you intended it? The mixture of 3- and 4-space indents tells me perhaps not, plus the fact that it repeats itself a bit ("It contains mock classes for ..." followed by "This module formerly contained mock classes for ...").

Also, will Sphinx handle `.mocks` and `.factories` correctly from this context? They're inside `sopel.tests`.

Reply:
It's close to what I intended, but not exactly that. Indeed, there is a repeat. Weird. I'll see to fix that, but the general idea is to say:

1. use the pytest plugin directly
2. or use sopel.tests' mocks and factories

For the `:mod:.mocks`, I don't know how, but yes, it works! My guess is that `sopel.test_tools` is documented within the `sopel.tests` namespace, and so Sphinx uses that as its reference point. It's weird, and I'll change that because it's way too confusing.